torch/csrc/jit/profiling_record.cpp

```diff
@@ -0,0 +1,118 @@
+#include <jit/profiling_record.h>
+
+namespace torch {
+namespace jit {
+
+void ProfilingRecord::addProfilingRecordAttribute(Node* pop) {
+  int64_t addr = 0; // if sizeof(ptr) == 4
+  addr = reinterpret_cast<int64_t>(this);
+  pop->i_(attr::storage, addr);
+}
+
+Node* ProfilingRecord::createProfileNode(Graph* graph, Value* o) {
+  Node* pop = graph->create(prim::profile, 0);
+  pop->i_(attr::data, reinterpret_cast<int64_t>(o));
+  addProfilingRecordAttribute(pop);
+  return pop;
+}
+
+void ProfilingRecord::instrumentBlock(Block* block) {
+  auto graph = block->owningGraph();
+  for (auto it = block->nodes().rend(); it != block->nodes().rbegin(); --it) {
+    auto n = *it;
+    for (auto o : n->outputs()) {
+      if (!o->type()->isSubclass(TypeKind::TensorType)) {
+        continue;
+      }
+      auto pop = createProfileNode(graph, o);
+      pop->addInput(o);
+      pop->insertAfter(n);
+    }
+
+    for (auto b : n->blocks()) {
+      instrumentBlock(b);
+    }
+  }
+}
+
+std::shared_ptr<Graph> ProfilingRecord::instrumentGraph(
+    const std::shared_ptr<Graph>& graph) {
+  auto new_g = graph->copy();
+  profiled_graph_ = new_g;
```

Comment:
weird -- I would not expect this method as it is named to delete other profiled graphs. Repeated calls to this method are pretty non-sensical and unsafe! What happens if we continue to record into the old profiled_graph_ after we switched it over.